system/lib/wasmfs/syscalls.cpp

```diff
@@ -1769,15 +1769,76 @@ int __syscall_fadvise64(int fd, off_t offset, off_t length, int advice) {
 }
 
 int __syscall__newselect(int nfds,
-                         intptr_t readfds_,
-                         intptr_t writefds_,
-                         intptr_t exceptfds_,
-                         intptr_t timeout_) {
-  // TODO: Implement this syscall. For now, we return an error code,
-  //       specifically ENOMEM which is valid per the docs:
-  //          ENOMEM Unable to allocate memory for internal tables
-  //          https://man7.org/linux/man-pages/man2/select.2.html
-  return -ENOMEM;
+                         intptr_t _readfds,
+                         intptr_t _writefds,
+                         intptr_t _exceptfds,
+                         intptr_t timeout) {
+  // Implement select in terms of `poll()`
+
+  // Part 1: convert select arguments into poll arguments
+  fd_set* readfds = (fd_set*)_readfds;
+  fd_set* writefds = (fd_set*)_writefds;
+  fd_set* exceptfds = (fd_set*)_exceptfds;
+
+  int n = 0;
+  struct pollfd* fds = (struct pollfd*)calloc(nfds, sizeof(struct pollfd));
+
+  for (int i = 0; i < nfds; i++) {
+    if (FD_ISSET(i, readfds) || FD_ISSET(i, writefds) || FD_ISSET(i, exceptfds)) {
```

Comment:
Does FD_ISSET check against the second parameter being null? I don't seem to see that in the code, so maybe we need to guard here?